dmscripts/generate_framework_agreement_data.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+import os
+from dmutils.apiclient.errors import HTTPError
+
+import sys
+if sys.version_info > (3, 0):
+    import csv
+else:
+    import unicodecsv as csv
+
+#  List of bad characters taken from: http://www.mtu.edu/umc/services/web/cms/characters-avoid/
+BAD_FILENAME_CHARACTERS = ['#', '%', '&', '{', '}', '\\', '<', '>', '*', '?', '/',
+                           '$', '!', "'", '"', ':', '@', '+', '`', '|', '=']
+
+
+class Supplier:
+
+    def __init__(self, declaration, lots):
+        if declaration[0] != lots[0]:
+            raise Exception("Supplier ID for lots does not match declaration")
+
+        self.supplier_id = declaration[0]
+
+        self.registered_company_name = declaration[20]
+        self.country_of_registration = declaration[26]
+        self.company_number = declaration[27]
+        self.registered_office_address = declaration[21]
+        self.contact_name = declaration[18]
+        self.contact_email = declaration[19]
+
+        self.lot1 = "Lot 1: Infrastructure as a Service (IaaS)" if int(lots[3]) > 0 else ""
+        self.lot2 = "Lot 2: Software as a Service (SaaS)" if int(lots[7]) > 0 else ""
+        self.lot3 = "Lot 3: Platform as a Service (PaaS)" if int(lots[5]) > 0 else ""
+        self.lot4 = "Lot 4: Specialist Cloud Services (SCS)" if int(lots[9]) > 0 else ""
+
+    def __str__(self):
+        return "ID: {}, RegName:{}, Country:{}, Num:{}, Addr:{}, Name:{}, email:{}".format(
+            self.supplier_id, self.registered_company_name, self.country_of_registration,
+            self.company_number, self.registered_office_address, self.contact_name, self.contact_email)
+
+
+def read_csv(filepath):
+    all_rows = []
+    with open(filepath, 'r') as csvfile:
+        csv_file = csv.reader(csvfile, delimiter=',', quotechar='"')
+        for row in csv_file:
+            all_rows.append(row)
+    return all_rows
+
+
+def make_filename_key(supplier_name, supplier_id):
+    sanitised_supplier_name = supplier_name.strip().replace(' ', '_').replace('&', 'and')
+    for bad_char in BAD_FILENAME_CHARACTERS:
+        sanitised_supplier_name = sanitised_supplier_name.replace(bad_char, '')
+    while '__' in sanitised_supplier_name:
+        sanitised_supplier_name = sanitised_supplier_name.replace('__', '_')
+    return "{}-{}".format(supplier_id, sanitised_supplier_name)
+
+
+def supplier_is_on_framework(client, supplier_id):
+    try:
+        framework_interest = client.get_supplier_framework_info(supplier_id, 'g-cloud-7')
+        if framework_interest['frameworkInterest']['onFramework']:
```

Comment:
`frameworkInterest` is correct (in that it's what the data API returns) but it shouldn't be.